### PR TITLE
MI-DPC and SystemPreview Class

### PR DESCRIPTION
**Pull request by xboldocky**
1. Adding additional system class with future preview of known variables.
2. Augmenting examples/control/part3 by adding preview of future references.
3. Adding an MI-DPC example.

### CODE REVIEW

**Comment by drgona** [comment]
@xboldocky, great job, almost there.

**both notebooks**
- add a single paragraph summary and conclusions final simulation results

**Part_6_mixed_integer_decisions.ipynb**
- Differentiable Mixed-Integer Optimal Control Problem markup cell seems to have white space and shows a long scroll
- move these before "Differentiable computational graph components" - create new markp cell to explan the rounding methods

```
if STE_method == 'sigmoid':
    def relaxed_round(x, slope=10.0): # differentiable nearest integer rounding via Sigmoid STE
        backward = (x-torch.floor(x)-0.5) # fractional value with rounding threshold
        return torch.round(x) + (torch.sigmoid(slope*backward) - torch.sigmoid(slope*backward).detach())
           #  forward pass↑     backward pass↑                                               no grad↑
    round_fn = lambda u_c, u_delta_relaxed: torch.cat((u_c, torch.clip(relaxed_round(u_delta_relaxed),u_delta_min,u_delta_max)), dim=-1)

elif STE_method == 'softmax':
    tau = 2 # temperature coefficient in (0,inf)
    round_fn = lambda u_c, u_delta_relaxed: torch.cat((u_c, gumbel_softmax(u_delta_relaxed, tau=tau, hard=True) @ integers.T), dim=-1)
```

**Part_3_ref_tracking_ODE.ipynb**
- move the ODE instantiation to its own cell call it ODE model before policy instantiations 
```
# white-box ODE model with no-plant model mismatch
two_tank_ode = ode.TwoTankParam()                   # ODE system equations implemented in PyTorch
two_tank_ode.c1 = nn.Parameter(torch.tensor(gt_model.c1), requires_grad=False)
two_tank_ode.c2 = nn.Parameter(torch.tensor(gt_model.c2), requires_grad=False)

# integrate continuous time ODE
integrator = integrators.RK4(two_tank_ode, h=torch.tensor(ts))   # using 4th order runge kutta integrator
```
- aggregate everything that is shared in shared cells, e.g., `loss = PenaltyLoss(objectives, constraints)`  is not specific to Problem without reference preview, so separate them
